### ISSUE TITLE
QuarkusUnitTest API to configure more than one .properties resource

### ISSUE THREAD

**Issue by rquinio1A** [quarkusio/quarkus#43914]
### Describe the bug

Following #43129 and with 999-SNAPSHOT, it's no longer possible to call multiple times QuarkusUnitTest#withConfigurationResource, only the last resource is taken into account.

With Quarkus 3.15.1, it was possible to do this: https://github.com/quarkiverse/quarkus-logging-splunk/blob/a7fa1ae74ef2b8f031e4cadb095f81545289e062/deployment/src/test/java/io/quarkiverse/logging/splunk/LoggingSplunkMinimalConfigTest.java#L22-L23

I guess it might have been a mis-usage of the API ? Is there another way to pass more than one .properties resource per QuarkusUnitTest ?

/cc @mkouba 

### Quarkus version or git rev

999-SNAPSHOT (master)

**Comment by mkouba** [comment]
> I guess it might have been a mis-usage of the API ? Is there another way to pass more than one .properties resource per QuarkusUnitTest ?

Well, the contract for `QuarkusUnitTest#withConfigurationResource()` was undefined before https://github.com/quarkusio/quarkus/pull/43181 so it's not a misuse per se ;-).

We could add a method that would load the properties from the given file and then use the `QuarkusUnitTest#overrideConfigKey()` for each property. It should be called `loadConfigProperties()` or something along these lines.

**Comment by rquinio1A** [comment]
Thanks. If `#withConfigurationResource` behavior will remain as-is and it's a new method, I guess short-term I'll refactor my tests anyway to use `#overrideConfigKey()` or merge the .properties, as I don't want to maintain a specific 3.15 LTS branch just for a test change if I can avoid it.

**Comment by mkouba** [comment]
> Thanks. If `#withConfigurationResource` behavior will remain as-is and it's a new method, I guess short-term I'll refactor my tests anyway to use `#overrideConfigKey()` or merge the .properties, as I don't want to maintain a specific 3.15 LTS branch just for a test change if I can avoid it.

👍 Sounds reasonable.

**Comment by radcortez** [comment]
You can always use `quarkus.config.locations` to add additional files.

**Comment by HerrDerb** [comment]
@radcortez While this does work:
```
  @RegisterExtension
    static final QuarkusUnitTest config = new QuarkusUnitTest()
            .addAdditionalDependency(ShrinkWrap.create(JavaArchive.class)
                    .addClasses(OidcTestApi.class)
                    .addAsManifestResource(EmptyAsset.INSTANCE, "beans.xml"))
            .withConfigurationResource("application-default.properties")
            .overrideConfigKey("quarkus.config.locations","application-oidc.properties");
```
, it does feel like a workaround.

**Comment by radcortez** [comment]
I understand the frustration about the API change, but I disagree that this is a workaround. The `quarkus.config.locations` is a well-supported property, and as the name says, it can be used to add additional configuration locations to load the configuration:
https://quarkus.io/guides/config-reference#configuration-reference

I would prefer to have that way rather than adding another way to set the configuration. I would be very much in favor of removing `withConfigurationResource` and `overrideConfigKey`. These methods were added when we didn't have profiled file-aware files, additional location support, etc. There are too many ways to set the configuration, making it harder to determine where the values are coming from.

Realistically, getting rid of `withConfigurationResource` and `overrideConfigKey` might be hard at this point. I'm not in favor of adding new methods, but I won't block it either.

**Comment by mkouba** [comment]
> I would be very much in favor of removing `withConfigurationResource` and `overrideConfigKey`.

👎 for removing `overrideConfigKey()`, it is a very convenient and transparent way to set a config property in a test. Much easier than adding a custom `application.properties` ShrinkWrap asset or even a dedicated file.

**Comment by HerrDerb** [comment]
`overrideConfigKey` is very convenient and makes the test conditions more readable (y)
I guess using `quarkus.config.locations` instead of a new method is acceptable for me. @mkouba you can decide if you would like to close my pr or add the method.

**Comment by mkouba** [comment]
Given the fact that all those "overriden" config properties end up in the same config source (`application.properties`) I don't think it's a problem to add one more method to `QuarkusUnitTest` which is still a semi-official API intended for extension authors.

@geoand @gsmet @cescoffier @maxandersen WDYT?

**Comment by radcortez** [comment]
> 👎 for removing `overrideConfigKey()`, it is a very convenient and transparent way to set a config property in a test. Much easier than adding a custom `application.properties` ShrinkWrap asset or even a dedicated file.

It would be my preference, but I also said that it would be hard to get rid of it :)

**Comment by geoand** [comment]
> QuarkusUnitTest which is still a semi-official API intended for extension authors

I agree, the target audience for these is limited so we don't have to be super duper strict about the API
